VideoLocker/src/main/java/org/edx/mobile/http/Api.java

```diff
@@ -153,39 +147,17 @@ public ProfileModel getProfile() throws Exception {
 
         logger.debug("Url for getProfile: " + urlWithAppendedParams);
 
-        String json = http.get(urlWithAppendedParams, getAuthHeaders()).body;
-
-        if (json == null) {
-            return null;
+        final HttpManager.HttpResult result = http.get(urlWithAppendedParams, getAuthHeaders());
+        if (result.statusCode < 200 || result.statusCode >= 300) {
```

Comment:
It's checking for a successful status code. Something which Retrofit does automatically. Perhaps I should also just delete this method and use the retrofit one? I think so...